### DISCUSSION TITLE
Add support for pagination/streaming

### DISCUSSION THREAD

**Discussion by geekingfrog** [awslabs/aws-sdk-rust#105] [General]
There are many requests in many services which returns paginated responses. For example, ListObject or ListBuckets in S3, Scan in DDB and so on.

It would be really nice to have a unified way to provide a method which returns a [futures::stream](https://docs.rs/futures/0.3.15/futures/stream/index.html), or I guess, a `TryStream` since the items are likely to be `Results`.

**Comment by rcoh** [comment]
Yes absolutely! We have an RFC design for paginators, would love your thoughts: https://github.com/awslabs/smithy-rs/discussions/346

**Reply by geekingfrog** [reply]
Oh excellent, I wasn't aware of that. I'll have a look (although I'm pretty noob)

**Comment by jmklix** [comment]
Created feature request [here](https://github.com/awslabs/aws-sdk-rust/issues/508) and locked this discussion
